mscore/albummanager.cpp

```diff
@@ -19,213 +24,621 @@
-    MasterScore* topScore = album->front();
+//---------------------------------------------------------
+//   addAlbumItem
+///     add the given AlbumItem to the AlbumManager
+///     creates the corresponding scoreList/View item
+///     the AlbumItem and Widget are saved in a new AlbumManagerItem
+//---------------------------------------------------------
 
+void AlbumManager::addAlbumItem(AlbumItem *albumItem)
+{
+    if (!albumItem) {
+        std::cout << "empty album item" << std::endl;
+        return;
+    }
     scoreList->blockSignals(true);
-    topScore->addMovement(score);
-    QString name = getScoreTitle(score);
-    QListWidgetItem* li = new QListWidgetItem(name, scoreList);
-    li->setFlags(Qt::ItemFlags(Qt::ItemIsSelectable | Qt::ItemIsEditable | Qt::ItemIsEnabled));
+    QString name = albumItem->score->title();
+    QTableWidgetItem* li = new QTableWidgetItem(name);
+    scoreList->setRowCount(scoreList->rowCount() + 1);
+    scoreList->setItem(scoreList->rowCount() - 1, 0, li);
+    li->setFlags(Qt::ItemFlags(Qt::ItemIsSelectable | Qt::ItemIsDragEnabled | Qt::ItemIsDropEnabled | Qt::ItemIsEnabled | Qt::ItemIsUserCheckable));
+    li->setCheckState(Qt::CheckState::Checked);
+    QTableWidgetItem* tid = new QTableWidgetItem("00:00:00");
+    scoreList->setItem(scoreList->rowCount() - 1, 1, tid);
+    tid->setFlags(Qt::ItemFlags(Qt::ItemIsEnabled));
+    AlbumManagerItem* albumManagerItem = new AlbumManagerItem(albumItem, li, tid);
+    _items.push_back(albumManagerItem);
     scoreList->blockSignals(false);
+    updateDurations();
+
+    // update the combined score to reflect the changes
+    if (tempScore) {
+        tempScore->addMovement(albumItem->score);
+        tempScore->update();
+        tempScore->doLayout(); // position the movements correctly
+        mscore->currentScoreView()->update(); // repaint
+    }
+}
+
+//---------------------------------------------------------
+//   updateDurations
+///     Calculates and updates (the labels of) the duration
+///     of the Album and of each individual score.
+//---------------------------------------------------------
 
-    topScore->setLayoutAll();
-    topScore->update();
+void AlbumManager::updateDurations()
+{
+    scoreList->blockSignals(true);
+    int seconds = 0;
+    int minutes = 0;
+    int hours = 0;
+
+    for (auto item : _items) {
+        bool temporarilyOpen = false;
+        if (item->albumItem->score == nullptr) {
+            item->albumItem->setScore(mscore->openScore(item->albumItem->_fileInfo.absoluteFilePath(), false));
+            temporarilyOpen = true;
+        }
+
+        if (item->albumItem->enabled) {
+            seconds += item->albumItem->score->duration();
+        }
+
+        int tempSeconds = item->albumItem->score->duration();
+        int tempMinutes = tempSeconds / 60;
+        tempSeconds -= tempMinutes * 60;
+        int tempHours = tempMinutes / 60;
+        tempMinutes -= tempHours * 60;
```

Comment:
Or, 

```cpp
int duration = item->albumItem->score->duration();
int tempSeconds =  duration % 60;
int tempMinutes = (duration / 60) % 60;
int tempHours = duration / 3600;
```
Not that that would make it better, it's just another way of doing it :)